### ISSUE TITLE
HeroRotation_Mage\Fire.lua since the patch 9.2 the same error as in 9.1 and 9.15

### ISSUE THREAD

**Issue by Hexus-Proud** [herotc/hero-rotation#842]
I get the same error every time since wow 9.1 with my Fire-Mage......

190x HeroRotation_Mage\Fire.lua:356: attempt to compare number with nil
[string "@HeroRotation_Mage\Fire.lua"]:356: in function <HeroRotation_Mage\Fire.lua:352>
[string "@HeroRotation_Mage\Fire.lua"]:870: in function `?'
[string "@HeroRotation\Main.lua"]:449: in function <HeroRotation\Main.lua:427>

Locals:
(*temporary) = 3.300000
(*temporary) = <function> defined @HeroRotation_Mage\Fire.lua:203
(*temporary) = 1
(*temporary) = nil
(*temporary) = false
(*temporary) = <table> {
 LastCastTime = 0
 LastHitTime = 0
 LastAppliedOnPlayerTime = 0
 LastRemovedFromPlayerTime = 0
 SpellName = "Flammenzuwachs"
 IsMelee = true
 SpellType = "Player"
 MaximumRange = 0
 MinimumRange = 0
 LastDisplayTime = 0
 SpellID = 337224
}
(*temporary) = "attempt to compare number with nil"
var_combustion_ready_time = 0
S = <table> {
 SunKingsBlessingBuff = <table> {
 }
 RemoveCurse = <table> {
 }
 MirrorsofTorment = <table> {
 }
 CombatMeditation = <table> {
 }
 HeatingUpBuff = <table> {
 }
 RadiantSpark = <table> {
 }
 FromTheAshes = <table> {
 }
 RingOfFrost = <table> {
 }
 BloodFury = <table> {
 }
 BerserkingBuff = <table> {
 }
 VolatileSolvent = <table> {
 }
 IcyPropulsion = <table> {
 }
 SoulIgnitionBuff = <table> {
 }
 AlexstraszasFury = <table> {
 }
 ArcaneExplosion = <table> {
 }
 Fireball = <table> {
 }
 ArcaneIntellectBuff = <table> {
 }
 ArcaneProdigy = <table> {
 }
 SunKingsBlessingBuffReady = <table> {
 }
 MirrorImage = <table> {
 }
 Meteor = <table> {
 }
 GroveInvigoration = <table> {
 }
 FlameAccretion = <table> {
 }
 IreOfTheAscended = <table> {
 }
 PustuleEruption = <table> {
 }
 BagofTricks = <table> {
 }
 WastelandPropriety = <table> {
 }
 FireBlast = <table> {
 }
 FlamePatch = <table> {
 }
 FrostNova = <table> {
 }
 SlowFall = <table> {
 }
 Invisibility = <table> {
 }
 ArcaneIntellect = <table> {
 }
 Kindling = <table> {
 }
 AncestralCall = <table> {
 }
 LightsJudgment = <table> {
 }
 Pyroblast = <table> {
 }
 EffusiveAnimaAccelerator = <table> {
 }
 BlazingBarrier = <table> {
 }
 SiphonedMalice = <table> {
 }
 ShiftingPower = <table> {
 }
 ExpandedPotentialBuff = <table> {
 }
 Combustion = <table> {
 }
 FieldOfBlossoms = <table> {
 }
 LivingBomb = <table> {
 }
 Pyroclasm = <table> {
 }
 Ignite = <table> {
 }
 BloodFuryBuff = <table> {
 }
 DoorofShadows = <table> {
 }
 InfernalCascade = <table> {
 }
 SearingTouch = <table> {
 }
 Fireblood = <table> {
 }
 RadiantSparkVulnerability = <table> {
 }
 InfernalCascadeBuff = <table> {
 }
 FirestormBuff = <table> {
 }
 GrislyIcicleDebuff = <table> {
 }
 PhoenixFlames = <table> {
 }
 HotStreakBuff = <table> {
 }
 Frostbolt = <table> {
 }
 Firestarter = <table> {
 }
 TomeofMonstruousConstructionsBuff = <table> {
 }
 PyroclasmBuff = <table> {
 }
 TimeWarp = <table> {
 }
 CombustionBuff = <table> {
 }
 Berserking = <table> {
 }
 Scorch = <table> {
 }
 Flamestrike = <table> {
 }
 AlterTime = <table> {
 }
 BlastWave = <table> {
 }
 RuneofPower = <table> {
 }
 DisciplinaryCommandBuff = <table> {
 }
 RuneofPowerBuff = <table> {
 }
 DragonsBreath = <table> {
 }
 GrislyIcicleBuff = <table> {
 }
 FlameOn = <table> {
 }
 Blink = <table> {
 }
 Counterspell = <table> {
 }
 SpellSteal = <table> {
 }
 FocusMagic = <table> {
 }
 IceBlock = <table> {
 }
 Fleshcraft = <table> {
 }
 Deathborne = <table> {
 }
 Soulshape = <table> {
 }
 RadiantSparkDebuff = <table> {
 }
 DeathborneBuff = <table> {
 }
}
var_kindling_reduction = 1
var_combustion_precast_time = nil
num = <function> defined @HeroRotation_Mage\Fire.lua:203
EnemiesCount8ySplash = 1
var_combustion_flamestrike = nil
var_combustion_cast_remains = nil
var_time_to_combustion = nil
var_firestarter_combustion = -1
max = <function> defined =[C]:-1
CovenantID = 3
DeathFathomEquipped = false
Player = <table> {
 CovenantID = <function> defined @HeroLib\Class\Unit\Player\Main.lua:58
 UseCache = true
 ChiDeficitPercentage = <function> defined @HeroLib\Class\Unit\Player\Power.lua:596
 GCDStartTime = <function> defined @HeroLib\Class\Unit\Player\Stat.lua:70
 Race = <function> defined @HeroLib\Class\Unit\Player\Main.lua:46
 Insanityrain = <function> defined @HeroLib\Class\Unit\Player\Power.lua:648
 FocusLossOnCastEnd = <function> defined @HeroLib\Class\Unit\Player\Power.lua:226
 PainMax = <function> defined @HeroLib

**Comment by Cilraaz** [comment]
It looks like you've somehow gotten into a scenario where somehow the CombustionTiming function is being called before the VarInit function. This would mean that when it tries to define a few timing variables in CombustionTiming, other variables that are being relied on are still nil. I've make a few changes that will hopefully prevent this from happening. Please try the changes in commits https://github.com/herotc/hero-rotation/commit/406beaf1e0aa9f8d5f0a6429fefebc299db0990f and https://github.com/herotc/hero-rotation/commit/cbaed74de8a09df0abe77aea6465c11ca4d6d162

**Comment by Cilraaz** [comment]
I did some more testing. It looks like this error should have only been happening if you open with a non-target spell (ground targeted, such as Flamestrike, or Arcane Explosion). In those instances, the variables were never being defined. I've changed when the VarInit function is called in commit https://github.com/herotc/hero-rotation/commit/cdb0760414ee08bf08302c2731dffb01f8823f76. This should be the final fix to the errors you were receiving.
